tests/update/patch_before_migration.pm

```diff
@@ -0,0 +1,97 @@
+# Patch SLE11* qcow2 images before migration (offline)
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Patch hdd's images before doing offline migration
+# Maintainer: Dumitru Gutu <dgutu@suse.de>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use registration;
+
+sub patching_sle11() {
+
+    my ($registration, $email, $regcode, $regcode_ha, $regcode_geo) = @_;
+    $email       = get_var("SCC_EMAIL");
+    $regcode     = get_var("SCC_REGCODE");
+    $regcode_ha  = get_var("SCC_REGCODE_HA");
+    $regcode_geo = get_var("SCC_REGCODE_GEO");
+
+    # block update process before registration
+    script_run("sed -i 's/true/false/g' /etc/PackageKit/PackageKit.conf");
+
+    if (get_var("SCC_REGCODE_GEO")) {
+        $registration = script_output("
+        suse_register -n -a email=$email -a regcode-sles=$regcode -a regcode-slehae=$regcode_ha -a regcode-slehaegeo=$regcode_geo", 90);
+    }
+    elsif (get_var("SCC_REGCODE_HA")) {
+        $registration = script_output("suse_register -n -a email=$email -a regcode-sles=$regcode -a regcode-slehae=$regcode_ha", 70);
+    }
+    else {
+        $registration = script_output("suse_register -n -a email=$email -a regcode-sles=$regcode", 90);
+    }
```

Comment:
how about setting the options in a string first, e.g. something like

```perl
my $regopts;
if (get_required_var('SCC_REGCODE_GEO')) {
    $regopts = "regcode-sles=$regcode -a regcode-slehae=$regcode_ha -a regcode-slehaegeo=$regcode_geo";
}
elsif (get_required_var('SCC_REGCODE_HA')) {
    $regopts = "regcode-sles=$regcode -a regcode-slehae=$regcode_ha";
}
else {
    $regopts = "regcode-sles=$regcode";
}
$registration = script_output("suse_register -n -a email=$email $regopts");
```

should make it a bit more readable and prevents that line break in line 31/32